django_migration_linter/sql_analyser/postgresql.py

```diff
@@ -32,6 +62,13 @@ class PostgresqlAnalyser(BaseAnalyser):
             mode=CheckMode.TRANSACTION,
             type=CheckType.WARNING,
         ),
+        Check(
+            code="CREATE_INDEX_EXCLUSIVE",
+            fn=has_create_index_in_transaction,
+            message="CREATE INDEX prolongs transaction, delaying lock release",
+            mode=CheckMode.TRANSACTION,
+            type=CheckType.ERROR,
```

Comment:
I think this is the main point I see with this contribution.

The check makes total sense, and thanks for adding it 👍 We definitely want to merge it :)

1/ It's not a silver bullet. I don't think by default we should raise an error during migration linting when doing this. To me, we should warn the user instead, that it could create a long-running locking transaction. Basically, stay in the same vein as the other indexing operations which also lock the table. It's the same risk => we know it will lock the table, but we can only warn that it is potentially a long-running lock, since we don't know about the real size and load of the table.

In short:
```suggestion
            type=CheckType.WARNING,
```
😁 

2/ It feels like a subset of `CREATE_INDEX`, since it checks `CREATE_INDEX` + being in a transaction + having an `ALTER TABLE` operation. I feel like there is room for something more generic, and not only when linked to index creation.
Basically, any combination of "transaction + exclusive lock operation + potential long-running operation" should trigger such a warning.
I'm completely fine starting with this specific check, we don't have to change the approach, but it's more to keep in mind that we could enhance this check to something more generic 🙂

Reply:
1. You're absolutely right to point out that this *could* have some false positives (i.e. indexing a small table will generally be fast & thus there's little cause for concern). I opted for an `ERROR` because I can't really think of many situations in which you would ever *need* to be creating an index within a transaction. I understand your rationale, though -- we should probably only call it an error if it will _always_ be the source of problems.

   I'll happily make this a `WARNING` and reach for `--warnings-as-errors` to elevate the warning for our use cases.

2. Also totally agreed that this could be made generic for detecting _any_ long-running queries that prolong lock release. I think the reason I targeted `CREATE INDEX` is that:
    - index creation is a very common migration operation
    - you can pretty much *always* move index creation out of the transaction. Other schema-modifying queries may not share that easy resolution.

   I would also be very happy to _start_ with this type of check, and replace it entirely with something more generic at a later time!
   
Thanks for the review & the thorough explanations! I'll be applying both your suggested changes, I think they'll help a ton.

Reply:
> I'll happily make this a WARNING and reach for --warnings-as-errors to elevate the warning for our use cases.

That would be great! Thanks a lot 🙇 
You can always use `--warnings-as-errors CREATE_INDEX_EXCLUSIVE` to mark only this one warning as an error. I think that should work 🤞 

> I would also be very happy to start with this type of check, and replace it entirely with something more generic at a later time!

Entirely agreed 👍  It was just an opening for the future, I'm completely fine with the current version, handling `CREATE INDEX [CONCURRENTLY]`.